src/Symfony/Component/Form/FormConfigBuilderInterface.php

```diff
@@ -13,15 +13,26 @@
 
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\PropertyAccess\PropertyPathInterface;
+use Symfony\Component\Form\Event\PreSetDataEvent;
+use Symfony\Component\Form\Event\PreSubmitEvent;
+use Symfony\Component\Form\Event\PostSetDataEvent;
+use Symfony\Component\Form\Event\SubmitEvent;
+use Symfony\Component\Form\Event\PostSubmitEvent;
 
 /**
+ * @template T
+ *
+ * @extends FormConfigInterface<T>
+ *
  * @author Bernhard Schussek <bschussek@gmail.com>
  */
 interface FormConfigBuilderInterface extends FormConfigInterface
 {
     /**
      * Adds an event listener to an event on this form.
      *
+     * @param FormEvents::* $eventName
+     * @param (callable(PreSetDataEvent<T>): void)|(callable(PostSetDataEvent<T>): void)|(callable(PreSubmitEvent): void)|(callable(SubmitEvent<T>): void)|(callable(PostSubmitEvent<T>): void) $listener
```

Comment:
Cannot we say the callable receive a `FormEvent`?

Reply:
No, because then you can't give a callable that accepts only one specific sub type.